api/spec/v1.yaml

```diff
@@ -1073,18 +1130,93 @@ components:
       description: |
         A list of consensus layer accounts.
     
+    AddressPreimage:
+      description: |
+        The data from which a consensus-style address (`oasis1...`, aka ed25519-backed,
+        aka staking address) was derived. Notably, for EVM runtimes like Sapphire,
+        this links the oasis address and the Ethereum address.
+
+        Oasis addresses are derived from a piece of data, such as an ed25519
+        public key or an Ethereum address. For example, [this](https://github.com/oasisprotocol/oasis-sdk/blob/b37e6da699df331f5a2ac62793f8be099c68469c/client-sdk/go/helpers/address.go#L90-L91)
+        is how an Ethereum is converted to an oasis address. The type of underlying data usually also
+        determines how the signatuers for this address are verified.
+        
+        Consensus supports only staking addresses (`context="oasis-core/address: staking"` below).
+        Runtimes support all types. This means that every consensus address is also
+        valid in every runtime. For example, in EVM runtimes, you can use staking
+        addresses, but only with oasis tools (e.g. a wallet); EVM contracts such as
+        ERC20 tokens or tools such as Metamask cannot interact with staking addresses.
+      type: object
+      properties:
+        context:
+          type: string
+          enum:
+            - "oasis-core/address: staking"
+            - "oasis-runtime-sdk/address: secp256k1eth"
+            - "oasis-runtime-sdk/address: sr25519"
+            - "oasis-runtime-sdk/address: multisig"
+            - "oasis-runtime-sdk/address: module"
+          description: |
+            The method by which the oasis address was derived from `address_data`.
+          example: "oasis-runtime-sdk/address: secp256k1eth"
+        context_version:
+          type: integer
+          nullable: true
+          default: 0
+          description: Version of the `context`.
+        address_data:
+          type: string
+          format: hex
+          description: |
+            The hex-encoded data from which the oasis address was derived. 
+            When `context = "oasis-runtime-sdk/address: secp256k1eth"`, this
+            is the Ethereum address (without the leading `0x`).
+          example: '9907A0cF64Ec9Fbf6Ed8FD4971090DE88222a9aC'
```

Comment:
it will lack the letter case checksum. plain hex encoding of the binary column only. not sure if that's uppercase or lowercase

Reply:
Good point! It's lowercase. Updated.